src/resolver.rs

```diff
@@ -2161,6 +2166,131 @@ impl FailForViolationConflict {
     }
 }
 
+async fn suggest_delta(
+    metadata: &cargo_metadata::Metadata,
+    network: Option<&Network>,
+    cache: &Cache,
+    package: &PackageNode<'_>,
+    failures: impl Iterator<Item = &SearchFailure>,
+) -> Option<DiffRecommendation> {
+    // Collect up the details of how we failed
+    struct Reachable<'a> {
+        from_root: SortedSet<&'a Option<VetVersion>>,
+        from_target: SortedSet<&'a Option<VetVersion>>,
+    }
+    let mut reachable = None::<Reachable<'_>>;
+    for SearchFailure {
+        reachable_from_root,
+        reachable_from_target,
+    } in failures
+    {
+        if let Some(Reachable {
+            from_root,
+            from_target,
+        }) = reachable.as_mut()
+        {
+            // FIXME: this is horrible but I'm tired and this avoids false-positives
+            // and duplicates. This does the right thing in the common cases, by
+            // restricting ourselves to the reachable nodes that are common to all
+            // failures, so that we can suggest just one change that will fix
+            // everything.
```

Comment:
This isn't that bad IMO :)

Reply:
Yeah, this isn't my comment, I just copied it from the original code.  It's fine.